### PR TITLE
Add DataSanitization hook to DataHandlerHook

### PR DESCRIPTION
**Pull request by mk-mxp**
Superseeds #443

Rewrote the hook using more generic naming. I am still testing if passing the reference is working this way, so maybe I need to change the HookProvider usage.

### CODE REVIEW

**Comment by mk-mxp** [comment]
Passing the reference didn't work, changed to passing backl the values as arrays.

**Comment by mk-mxp** [comment]
@oliverklee Please take a look.

**Comment by oliverklee** [comment]
@mk-mxp Thanks for the PR! I'll have a look.

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 1]
```diff
@@ -823,3 +824,50 @@ Implementieren Sie die benötigten Methoden gemäß dem Interface:
...
         }
     }
+
+.. _datasanitization_de:
+
+Hooks zur Datenbereinigung bei der TCE-Validierung
+""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Es gibt einen Hook in den Data Handler, um bei der TCE-Validierung (vor dem Speichern eines
+Seminars) zusätzliche Bedingungen abzuprüfen und eigene dynamische Anpassungen der Daten
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst:834]: eines Seminars -> einer Veranstaltung

Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst:834]: abzuprüfen -> zu prüfen

File: Documentation/DE/Referenz/Hooks/Index.rst [hunk 2]
```diff
@@ -823,3 +824,50 @@ Implementieren Sie die benötigten Methoden gemäß dem Interface:
...
+Das Verfahren der TCE-Validierung ist von Typo3 vorgegeben. `seminars` erhält dabei die Formular-Daten
+aus dem FlexForm des Content-Elements und speichert nötige Änderungen der eingetragenen Werte in die
+Datenbank.
+
+Ihre Klasse, die :php:`\OliverKlee\Seminars\Hooks\Interfaces\DataSanitization` implementiert,
+machen Sie seminars in :file:`ext_localconf.php` Ihrer Extension bekannt:
+
+.. code-block:: php
+
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][\OliverKlee\Seminars\Hooks\Interfaces\DataSanitization:class][]
```
Comment (oliverklee) [inline Documentation/DE/Referenz/Hooks/Index.rst:846]: `:` -> `::`

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 3]
```diff
@@ -822,3 +823,49 @@ Implement the methods required by the interface:
...
+.. _datasanitization_en:
+
+Hooks for the data sanitization on TCE validation
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+There is a hook into the data handler to additionaly manipulate `seminars` FlexForm data during
+TCE validation (just before storing the data). You may apply additional constraints and dynamically
+adjust values (e.g. registration deadline = begin date - 14 days).
+
+TCE validation is a Typo3 defined process. `seminars` gets the form values from the content element's
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst:836]: Typo3 defined -> TYPO3-defined

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 4]
```diff
@@ -822,3 +823,49 @@ Implement the methods required by the interface:
...
+FlexForm and stores required changes of the values into the DB.
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst:837]: DB -> database

File: Documentation/EN/Reference/Hooks/Index.rst [hunk 5]
```diff
@@ -822,3 +823,49 @@ Implement the methods required by the interface:
...
+
+Register your class that implements :php:`\OliverKlee\Seminars\Hooks\Interfaces\DataSanitization`
+like this in :file:`ext_localconf.php` of your extension:
+
+.. code-block:: php
+
+    $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][\OliverKlee\Seminars\Hooks\Interfaces\DataSanitization:class][]
```
Comment (oliverklee) [inline Documentation/EN/Reference/Hooks/Index.rst:844]: `:` -> `::`

File: Classes/Hooks/DataHandlerHook.php [hunk 6]
```diff
@@ -121,10 +121,11 @@ private function processSingleEvent(int $uid)
         $this->copyDatesFromTimeSlots($uid, $updatedData);
         $this->sanitizeEventDates($updatedData);
 
+        /** @var HookProvider */
```
Comment (oliverklee) [inline Classes/Hooks/DataHandlerHook.php:124]: `@var` annotations for local variables always need the variable name (unlike the annotation for fields):

`/** @var HookProvider $dataSanitizationHookProvider*/`

**Pull request merged**